crypto/src/main/java/org/hyperledger/besu/crypto/KeyPairUtil.java

```diff
@@ -32,6 +32,7 @@
 
 public class KeyPairUtil {
   private static final Logger LOG = LogManager.getLogger();
+  private static final EllipticCurveSignature ELLIPTIC_CURVE_SIGNATURE = EllipticCurveSignatureFactory.getInstance();
```

Comment:
Fixing this as a static final field makes runtime or startup determination of the curve problematic.  The config would be locked in at _class loading_ time.  We should use the method on the factory instead.  Locking the factory is one thing, locking what the factory produces is something entirely different.

If we had a lazy initialized value that would also allow for later than classloading configuration. `com.google.common.base.Suppliers#memoize` is a quick path to victory here.

Reply:
Thank you for the tip. I've used now `Suppliers#memoize` instead.